File: 04/exercise1-objc/die.m
#import "die.h"
#import <stdlib.h>

@interface die()

@end

@implementation die

- (void)roll {
	self.point = rand() % self.sides + 1;
}

- (id) init {
	self = [super init];
	self.sides = 6;
	return self;
}

- (id) initWithSidesOverFour: (int)sides {
	self = [super init];
	if(sides>=4) {
		self.sides = sides;
	} else {
		printf("WARNING: The value of side should be greater or equal to 4.\n");
		self.sides = sides;
	}
	return self;
}

@end

File: 04/die.cpp
#include "die.h"

void die::roll(void) {
	point_ = rand() % sides_ + 1;
}
		
int die::getValue(void) {
	return point_;
}
		
int die::getNumsides(void) {
	return sides_;
}

die::die() {
	sides_ = 6;
}

die::die(int numSides) {
	if(numSides>=4){
		sides_ = numSides;
	} else {
		cout << "WARNING: The value of side should be greater or equal to 4." << endl;
		sides_ = numSides;
	}
}

die::~die() {
	
}